crates/router/src/connector/payu/transformers.rs

```diff
@@ -0,0 +1,550 @@
+                },
+            )?,
+            pay_methods: payment_method,
+            continue_url: None,
+        })
+    }
+}
+
+pub struct PayuAuthType {
+    pub(super) api_key: String,
+    pub(super) merchant_pos_id: String,
+}
+
+impl TryFrom<&types::ConnectorAuthType> for PayuAuthType {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+        match auth_type {
+            types::ConnectorAuthType::BodyKey { api_key, key1 } => Ok(Self {
+                api_key: api_key.to_string(),
+                merchant_pos_id: key1.to_string(),
+            }),
+            _ => Err(errors::ConnectorError::FailedToObtainAuthType)?,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum PayuPaymentStatus {
+    Success,
+    #[serde(rename = "WARNING_CONTINUE_REDIRECT")]
+    WarningContinueRedirect,
+    #[serde(rename = "WARNING_CONTINUE_3DS")]
+    WarningContinue3ds,
+    #[serde(rename = "WARNING_CONTINUE_CVV")]
+    WarningContinueCvv,
+    #[default]
+    Pending,
+}
+
+impl From<PayuPaymentStatus> for enums::AttemptStatus {
+    fn from(item: PayuPaymentStatus) -> Self {
+        match item {
+            PayuPaymentStatus::Success => Self::Pending,
+            PayuPaymentStatus::WarningContinue3ds => Self::Pending,
+            PayuPaymentStatus::WarningContinueCvv => Self::Pending,
+            PayuPaymentStatus::WarningContinueRedirect => Self::Pending,
+            PayuPaymentStatus::Pending => Self::Pending,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct PayuPaymentsResponse {
+    pub status: PayuPaymentStatusData,
+    pub redirect_uri: String,
+    pub iframe_allowed: Option<bool>,
+    pub three_ds_protocol_version: Option<String>,
+    pub order_id: String,
+    pub ext_order_id: Option<String>,
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, PayuPaymentsResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ParsingError>;
+    fn try_from(
+        item: types::ResponseRouterData<F, PayuPaymentsResponse, T, types::PaymentsResponseData>,
+    ) -> Result<Self, Self::Error> {
+        // todo 3ds
+        // let mut base_url = item.response.redirect_uri.clone();
+        // base_url.set_query(None);
+        // let redirection_data = Some(services::RedirectForm {
+        //     url: base_url.to_string(),
+        //     method: services::Method::Get,
+        //     form_fields: std::collections::HashMap::from_iter(
+        //         item.response
+        //             .redirect_uri
+        //             .query_pairs()
+        //             .map(|(k, v)| (k.to_string(), v.to_string())),
+        //     ),
+        // });
```

Comment:
Is this commented code snippet required?

If yes, could you please create an issue for the same and include the issue number here?